polymath/polynomial.py

```
@@ -487,26 +604,65 @@ def eval(self, x, recursive=True):
+
+                # Use example= to copy properties, but still need arg for the values
+                return Scalar(const_values, mask=None, derivs=derivs, example=self)
             else:
-                return Scalar(example=self.wod)
+                # Use example=self.wod to copy properties without derivatives
+                return Scalar(const_values, mask=None, derivs={}, example=self.wod)
 
         x = Scalar.as_scalar(x, recursive=recursive)
         x_powers = [1., x]
         x_power = x
         for k in range(1, self.order):
```

Ruff (0.14.7)

Loop control variable `k` not used within loop body

Rename unused `k` to `_k`

(B007)